data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/ParseTreeCoercionServiceTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class ParseTreeCoercionServiceTest {
+
+    @Mock
+    private TerminalNode terminalNode;
+
+    @Mock
+    private Token token;
+
+    private final ParseTreeCoercionService objectUnderTest = new ParseTreeCoercionService();
+
+    @BeforeEach
+    void setUp() {
+        lenient().when(terminalNode.getSymbol()).thenReturn(token);
+    }
+
+    @Test
+    void testCoerceTerminalNodeStringType() throws ParseTreeCoercionException {
+        when(token.getType()).thenReturn(DataPrepperExpressionParser.String);
+        final String testString = "test string";
+        when(terminalNode.getText()).thenReturn(testString);
+        final Event testEvent = createTestEvent(new HashMap<>());
+        final Object result = objectUnderTest.coerceTerminalNode(terminalNode, testEvent);
+        assertThat(result, instanceOf(String.class));
+        assertThat(result, equalTo(testString));
+    }
+
+    @Test
+    void testCoerceTerminalNodeIntegerType() throws ParseTreeCoercionException {
+        when(token.getType()).thenReturn(DataPrepperExpressionParser.Integer);
+        final Integer testInteger = 1;
```

Comment:
I strongly recommend the use of random numbers here (within a decent range). It is quite easy to write against a test with a value like `1`. 

Alternatively, you can make this a `@ParameterizedTest` and verify against a few values.

Reply:
Good suggestiong @dlvenable!

As a best practice, to guarantee consistent test results, I recommend avoiding random values in test cases. Either method is acceptable in this use case and both are used throught Data Preppers test code.